hazelcast-hibernate/hazelcast-hibernate3/src/test/java/com/hazelcast/hibernate/HibernateStatisticsTestSupport.java

```diff
@@ -223,6 +269,41 @@ protected DummyEntity executeQuery(SessionFactory factory, long id) {
         }
     }
 
+    protected List<DummyEntity> executeListQueryInTransaction(SessionFactory factory) {
+        Session session = factory.openSession();
+        Transaction txn = null;
+        List<DummyEntity> res = null;
+        try {
+            txn = session.beginTransaction();
+            Query query = session.createQuery("FROM DummyEntity");
+            query.setCacheable(true);
+
+            res= query.list();
+            txn.commit();
+        } finally {
+            session.close();
```

Comment:
It can now happen that if an exception is thrown, the transaction is not aborted. Is this intentional?